components/telemetry-operator/internal/configbuilder/output.go

```diff
@@ -0,0 +1,131 @@
+package configbuilder
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	telemetryv1alpha1 "github.com/kyma-project/kyma/components/telemetry-operator/apis/telemetry/v1alpha1"
+	"github.com/kyma-project/kyma/components/telemetry-operator/internal/utils/envvar"
+)
+
+func createOutputSection(pipeline *telemetryv1alpha1.LogPipeline, config PipelineConfig) string {
+	output := &pipeline.Spec.Output
+	if len(output.Custom) > 0 {
+		return generateCustomOutput(output, config.FsBufferLimit, pipeline.Name)
+	}
+
+	if output.HTTP.Host.IsDefined() {
+		return generateHTTPOutput(&output.HTTP, config.FsBufferLimit, pipeline.Name)
+	}
+
+	if output.Loki.URL.IsDefined() {
+		return generateLokiOutput(&output.Loki, config.FsBufferLimit, pipeline.Name)
+	}
+
+	return ""
```

Comment:
I think we need to return an error here since we require an output. Before we had `fmt.Errorf("output plugin not defined")` in the old `config_builder`. Then `MergeSectionsConfig` also doesn't return `nil` all the time.

Reply:
See below